### PR TITLE
add doc on required for nested eager loading

### PR DESCRIPTION
**Pull request by syldor**
Infos on the `required` option when doing nested eager loading.

### CODE REVIEW

File: docs/docs/models-usage.md [hunk 1]
```diff
@@ -534,6 +534,19 @@ User.findAll({
   */
 })
 ```
+This will produce an outer join. However, a `where` clause on a related model will create an inner join and return only the instances that have matching sub-models. To return all the instances, you should add `required: false`.
```
Comment (janmeier) [inline docs/docs/models-usage.md]: Not sure it makes sense to add this under nested eager laoding - it should probably be higher up

File: docs/docs/models-usage.md [hunk 2]
```diff
@@ -534,6 +534,19 @@ User.findAll({
...
+
+```js
+User.findAll({
+  include: [
+    {model: Tool, as: 'Instruments', include: [
+      {model: Teacher, include: [ /* etc */], required: false}
```
Comment (janmeier) [inline docs/docs/models-usage.md]: There is no where clause here so required is actually redundant

File: docs/docs/models-usage.md (hunk 1 above)
Comment (mickhansen) [inline docs/docs/models-usage.md]: It's better than nothing, so just merging :D

File: docs/docs/models-usage.md (hunk 2 above)
Comment (syldor) [inline docs/docs/models-usage.md]: dammit ! I forgot the where clause

**Pull request merged**